### PR TITLE
tag TrafficDistribution tests to avoid running on clusters without th…

### PR DESCRIPTION
**Pull request by aojea**
…e feature enabled

/kind bug
/kind failing-test
```release-note
NONE
```

Detected by @soltysh on the openshift CI, we didn't detected because we don't run e2e tets on clusters with the feature disabled, and the only multizone tests we run are in  https://testgrid.k8s.io/sig-network-kind#sig-network-kind,%20multizone&include-filter-by-regex=TrafficDistribution

### CODE REVIEW

**Review by soltysh — APPROVED** [review]
/lgtm
/approve

**Comment by aojea** [comment]
we need api owners 
Tim or Jordan
/assign @liggitt

File: staging/src/k8s.io/api/core/v1/types.go [hunk 1]
```diff
@@ -5368,6 +5368,8 @@ type ServiceSpec struct {
 	// not set, the implementation will apply its default routing strategy. If set
 	// to "PreferClose", implementations should prioritize endpoints that are
 	// topologically close (e.g., same zone).
+	// This is an alpha field and requires enabling ServiceTrafficDistribution feature.
```
Comment (thockin) [inline staging/src/k8s.io/api/core/v1/types.go:5371]: need to update-codegen

**Comment by aojea** [comment]
> Kubernetes e2e suite: [It] [sig-network] Services should create endpoints for unready pods 

known flake

**Comment by aojea** [comment]
@thockin updated

**Review by thockin — COMMENTED** [review]
Thanks!

/lgtm
/approve

**Pull request merged**